mail_filter_adressee_by_contact/models/mail_compose_message.py

```diff
@@ -23,13 +23,24 @@ class MailComposer(models.TransientModel):
         "Additional Contacts",
     )
 
-    apply_filter = fields.Boolean(default=True, string="Filtering relevant adressees")
+    apply_filter = fields.Selection(
+        (
+            [
+                ("contacts", "Contacts"),
+                ("users", "Users"),
+                ("all", "All"),
+            ]
+        ),
+        string="Filtering relevant adressees",
+        default="contacts",
+        required=True,
+    )
 
     @api.onchange("apply_filter")
     def _get_partner_ids_domain(self):
         domain = [("type", "!=", "private")]
         model = self._context.get("active_model")
-        if model in FILTERED_MODEL and self.apply_filter:
+        if model in FILTERED_MODEL and self.apply_filter == "contacts":
```

Comment:
```
if self.apply_filter:
   method_name = f"_get_domain_for_{model.replace('.', '_')}"
   if hasattr(self, method_name):
       domain = getattr(self, method_name)
```

Reply:
I changed it

Reply:
No need to check if model is in FILTERED_MODEL, just check if model exists and you can remove FILTERED_MODEL

```
if model and self.apply_filter:
    # Other code
```